core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -180,7 +180,15 @@ class CityInfo {
             val amount = getTileResourceAmount(tileInfo) * civInfo.getResourceModifier(resource)
             if (amount > 0) cityResources.add(resource, amount, "Tiles")
         }
+        for (tileInfo in getTiles().filter { it.improvement!= null }) {
+            val tileImprovement = tileInfo.getTileImprovement() ?: continue
```

Comment:
If you've filtered the tiles to only the tiles that have improvements, this will never be null so there's no need for ?: continue

Reply:
i thought that, but android studio complains if there's no null check